lib/util/fs/index.ts

```diff
@@ -148,7 +188,11 @@ export async function findLocalSiblingOrParent(
  */
 export async function readLocalDirectory(path: string): Promise<string[]> {
   const { localDir } = GlobalConfig.get();
-  const localPath = upath.join(localDir, path);
+  const localPath = upath.resolve(localDir, path);
+  if (!isPathInBaseDir(localPath, localDir)) {
+    return Promise.reject();
```

Comment:
maybe better throw an exception? i think the code which is using this function expects an error object, so we will fail with possible null ref in catch blocks.

Reply:
Hmm, either my search is faulty or nobody is using `readLocalDirectory()`? It was introduced with PR https://github.com/renovatebot/renovate/pull/10469 and commit https://github.com/renovatebot/renovate/commit/eb8ad0c38ebce9dc580a9c9a9819dd1b54bd2ad3 but I don't see where it would actually be used, apart from tests.

Remove dead code or turn `Promise.reject()` into `throw new Error()`? Please let me know what you'd prefer 👍

Reply:
remove if it's unused